src/terrain/services/qms.clj

```diff
@@ -1,9 +1,16 @@
 (ns terrain.services.qms
   (:require [clojure.set :refer [difference]]
             [clojure-commons.core :refer [remove-nil-values]]
+            [clojure-commons.exception-util :as cxu]
             [terrain.clients.iplant-groups.subjects :as subjects]
             [terrain.clients.qms :as qms]))
 
+(defn- validate-username
+  "Throws an error if a user with the given username doesn't exist."
+  [username]
+  (when (empty? (:subjects (subjects/lookup-subjects [username])))
+    (cxu/bad-request (str "user '" username "' does not exist"))))
```

Comment:
Calling `cxu/bad-request` has the unfortunate effect of formatting the error message slightly differently than it would be formatted when the error happened in QMS, but it's more consistent with other Terrain endpoints. I'm planning to add some code to reformat errors that are detected in QMS so that they're consistent with errors that are detected in our Clojure services.